tests/e2e/crud/crud_test.go

```diff
@@ -318,3 +321,339 @@ func TestE2EStandardCRUD(t *testing.T) {
 		t.Fatalf("an error occurred: %s", err)
 	}
 }
+
+func TestE2ECRUDWithSkipStrictExistCheck(t *testing.T) {
+	t.Cleanup(teardown)
+	ctx := context.Background()
+
+	op, err := operation.New(host, port)
+	if err != nil {
+		t.Fatalf("an error occurred: %s", err)
+	}
+
+	// #1 run Update with SkipStrictExistCheck=true and check that it fails.
+	err = op.UpdateWithParameters(
+		t,
+		ctx,
+		operation.Dataset{
+			Train: ds.Train[updateFrom : updateFrom+updateNum],
+		},
+		true,
+		1,
+		func(t *testing.T, status int32, msg string) error {
+			t.Helper()
+
+			if status != int32(codes.NotFound) {
+				return errors.Errorf("the returned status is not NotFound on Update #1: %s", err)
+			}
+
+			t.Logf("received a NotFound error on #1: %s", msg)
+
+			return nil
+		},
+		func(t *testing.T, err error) error {
+			t.Helper()
+
+			st, _, _ := status.ParseError(err, codes.Unknown, "")
+			if st.Code() != codes.NotFound {
+				return err
+			}
+
+			return nil
+		},
+	)
+	if err != nil {
+		t.Fatalf("an error occurred: %s", err)
+	}
+
+	// #2 run Update with SkipStrictExistCheck=false, and check that the internal Remove Operation returns a NotFound error
+	err = op.UpdateWithParameters(
+		t,
+		ctx,
+		operation.Dataset{
+			Train: ds.Train[updateFrom : updateFrom+updateNum],
+		},
+		false,
+		1,
+		func(t *testing.T, status int32, msg string) error {
+			t.Helper()
+
+			if status != int32(codes.NotFound) {
+				return errors.Errorf("the returned status is not NotFound on Update #2: %s", err)
+			}
+
+			t.Logf("received a NotFound error on #2: %s", msg)
+
+			return nil
+		},
+		func(t *testing.T, err error) error {
+			t.Helper()
+
+			// TODO: This should be NotFound error but it returns
+			// `code = Unknown desc = ngt uuid 7's object not found: ...`
+			// st, _, _ := status.ParseError(err, codes.Unknown, "")
+			// if st.Code() != codes.NotFound {
+			// 	return err
+			// }
```

Comment:
There're several TODOs like this. They cannot resolve now because they are caused by bugs in LB Gateway / Agent handlers.